### PR TITLE
New jupyter notebooks for exchange contract family

### PR DESCRIPTION
**Pull request by cmickeyb**
ignificant update for the Jupyter notebooks in the
exchange contract family. Notebooks added for creating
and importing orders, for interacting with an issuer
as an account holder, and for kick starting the factories
there is a new index file.

To make all this work, this update adds an experimental
file format for sharing contract information. The file
uses the python zip library (distributed with python)
to package the PDO files and the context files.

### CODE REVIEW

File: exchange-contract/docs/notebooks/factories/token-issuer.ipynb [hunk 1]
```diff
@@ -107,29 +107,15 @@
...
-    "    import papermill as pm\n",
-    "    pm.execute_notebook(\n",
-    "        template_file,\n",
-    "        instance_file,\n",
-    "        prepare_only=True,\n",
-    "        parameters=instance_parameters,\n",
-    "    )\n",
-    "instance_path = os.path.join(os.pardir, 'instances', token_class, 'issuer.ipynb')\n",
-    "ip_display.display(ip_display.Markdown('[Token Issuer Notebook]({})'.format(instance_path)))"
+    "instance_file = ex_jupyter.instantiate_notebook_from_template(token_class, 'token-issuer', parameters)   \n",
```
Comment (prakashngit) [inline exchange-contract/docs/notebooks/factories/token-issuer.ipynb:117]: parameters must be instance_parameters ?

File: exchange-contract/docs/notebooks/index.ipynb [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+   "metadata": {},
+   "source": [
+    "### Simplified configuration\n",
+    "\n",
+    "If you are running the PDO services in the common test configuration (e.g. with 5 provisioning services, 5 enclave service, and 5 storage services all running on the same host, you can create all of the necessary configuration files with the command:\n",
+    "    \n",
+    "```\n",
+    "    $ export service_host=<service hostname>\n",
+    "    $ ${PDO_INSTALL_ROOT}/bin/pdo-create-service-groups.psh --service_host ${service_host}\n",
+    "```\n",
```
Comment (prakashngit) [inline exchange-contract/docs/notebooks/index.ipynb:40]: I guess you meant $PDO_INSTALL_ROOT/bin/pdo-shell $PDO_HOME/bin/pdo-create-service-groups.psh --service_host $service_host to create the groups file ?

File: exchange-contract/docs/notebooks/index.ipynb [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+   "id": "413968fd-83db-44bc-869b-a711a671f839",
+   "metadata": {},
+   "source": [
+    "### Create a services database\n",
+    "\n",
+    "The PDO client requires information about the enclave services that will be used for contracts. Typically, a service provider will create a site description file. For example, if the site description file is called `site.toml` (the default), then the following command will add all of the services to the database:\n",
+    "\n",
+    "```\n",
+    "$ pdo-services-db import --file site.toml\n",
+    "```"
```
Comment (prakashngit) [inline exchange-contract/docs/notebooks/index.ipynb:68]: pdo-services-db -> pdo-service-db.

Also, may be add the full path? $PDO_INSTSALL_ROOT/_dev/bin/pdo-service-db import --file site.toml

File: exchange-contract/docs/notebooks/factories/token-issuer.ipynb [hunk 4]
```diff
@@ -107,29 +107,15 @@
...
    "outputs": [],
    "source": [
-    "instance_directory = os.path.join(notebook_directory, 'instances', token_class)\n",
-    "instance_file = os.path.join(instance_directory, 'issuer.ipynb')\n",
-    "template_file = os.path.join(notebook_directory, 'templates', 'token-issuer.ipynb')\n",
+    "instance_parameters = {\n",
+    "    'token_owner' : identity,\n",
+    "    'token_class' : token_class,\n",
+    "    'service_host' : service_host,\n",
+    "    'notebook_directory' : notebook_directory,\n",
```
Comment (prakashngit) [inline exchange-contract/docs/notebooks/factories/token-issuer.ipynb:114]: this is really a comment on the "Configure Token information" Cell. Given that we already specify service_host as ENV, do we still need service_host parameter to be explicitly specified? 

Also, there is something wierd with this Configure Token information Cell where you enter all the info. Before this PR, after the last info (service_host) when I press ENTER, the execution cursor automatically goes to the next cell, (as expcted), but now, it comes back to the same cell, and when i press "play button", i am asked to enter all the info all over again. Of course, I can manually move the play button to the next cell, that works.

File: exchange-contract/docs/notebooks/index.ipynb (hunk 2 above)
Comment (cmickeyb) [reply exchange-contract/docs/notebooks/index.ipynb:40]: you can execute directly

**Comment by cmickeyb** [comment]
Updated in #17.

**Pull request closed without merging**